.github/workflows/publish-pypi-deploy.yml

```diff
@@ -19,10 +21,10 @@ jobs:
       id-token: write
     steps:
     - name: Checkout
-      uses: actions/checkout@v5
+      uses: actions/checkout@08c6903cd8c0fde910a37f88322edcfb5dd907a8  # v5.0.0
```

Comment:
How ugly... but well, security needs...
That also means that the dependabot for github actions will not work anymore, so it probably makes sense to remove that from the `dependabot.yml`.

Reply:
I hadn't considered that, but [this issue](https://github.com/dependabot/dependabot-core/issues/4691) seems to point to dependabot being able to handle it.

Reply:
I think the update check works, but not security alerts